src/servers/grpc_server.cc

```diff
@@ -2188,7 +2188,10 @@ InferHandler<
   auto barrier = std::make_shared<Barrier>(2);
 
   thread_.reset(new std::thread([this, barrier] {
-    StartNewRequest();
+    size_t registered_infer_request_count = 2;
```

Comment:
Can you define REGISTER_GRPC_INFER_REQUEST_COUNT as a macro at the top and then use that here?
Also, does it make sense to expose this as a cli option for the tritonserver executable?